pandas/core/groupby/generic.py

```diff
@@ -1309,42 +1309,51 @@ def _transform_general(self, func, *args, **kwargs):
         gen = self.grouper.get_iterator(obj, axis=self.axis)
         fast_path, slow_path = self._define_paths(func, *args, **kwargs)
 
-        for name, group in gen:
-            object.__setattr__(group, "name", name)
-
-            # Try slow path and fast path.
-            try:
-                path, res = self._choose_path(fast_path, slow_path, group)
-            except TypeError:
-                return self._transform_item_by_item(obj, fast_path)
-            except ValueError as err:
-                msg = "transform must return a scalar value for each group"
-                raise ValueError(msg) from err
-
+        def process_result(applied, group, res):
```

Comment:
can this be changed to return the object to be appended to applied? rather than mutating